purchases/src/main/kotlin/com/revenuecat/purchases/common/Dispatcher.kt

```diff
@@ -56,7 +60,10 @@ internal open class Dispatcher(
         synchronized(this.executorService) {
             if (!executorService.isShutdown) {
                 val future = if (delay != Delay.NONE && executorService is ScheduledExecutorService) {
-                    val delayToApply = (delay.minDelay.inWholeMilliseconds..delay.maxDelay.inWholeMilliseconds).random()
+                    var delayToApply = (delay.minDelay.inWholeMilliseconds..delay.maxDelay.inWholeMilliseconds).random()
+                    if (runningIntegrationTests) {
+                        delayToApply = (delayToApply * INTEGRATION_TEST_DELAY_PERCENTAGE).toLong()
+                    }
```

Comment:
This is not ideal... Honestly it was cleaner before, but I think it's worth this for the tradeoff of removing the extra flavor... Lmk what you think!

Reply:
I honestly think the other way was a bit "magic" in the sense that it was hard to know the override was happening, I think this way it's more explicit